### PR TITLE
Copy header from the origin header in dead letter queue

### PR DESCRIPTION
**Pull request by loicmathieu**

### CODE REVIEW

**Comment by loicmathieu** [comment]
I didn't add any test because I'm not sure where to put it.
Also, I'm not fan of the new String creation, if you have better solution (maybe via a Buffer) please provide some advice.

**Review by cescoffier — APPROVED** [review]

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/fault/KafkaDeadLetterQueue.java [hunk 1]
```diff
@@ -102,6 +102,7 @@ private String getThrowableMessage(Throwable throwable) {
         dead.addHeader(DEAD_LETTER_TOPIC, record.getTopic());
         dead.addHeader(DEAD_LETTER_PARTITION, Integer.toString(record.getPartition()));
         dead.addHeader(DEAD_LETTER_OFFSET, Long.toString(record.getOffset()));
+        record.getHeaders().forEach(header -> dead.addHeader(header.key(), new String(header.value())));
```
Comment (cescoffier) [inline smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/fault/KafkaDeadLetterQueue.java:105]: We can introduce huge string creation and even invalid strings
Can you switch to: 

```
record.getHeaders().forEach(header -> dead.addHeader(KafkaHeader.header(header.key(), Buffer.buffer(header.value()))));
```

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/fault/KafkaDeadLetterQueue.java (hunk 1 above)
Comment (loicmathieu) [reply smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/fault/KafkaDeadLetterQueue.java]: done

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
Thanks!

**Pull request merged**